src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -3081,11 +3096,16 @@ public boolean generatedKeyAlwaysReturned() throws SQLException {
     throw new SnowflakeLoggedFeatureNotSupportedException(session);
   }
 
+  // unchecked
   @Override
   public <T> T unwrap(Class<T> iface) throws SQLException {
-    logger.debug("public <T> T unwrap(Class<T> iface)");
+    logger.debug("<T> T unwrap(Class<T> iface)");
 
-    throw new SnowflakeLoggedFeatureNotSupportedException(session);
+    if (!iface.isInstance(this)) {
```

Comment:
Add ability to unwrap SnowflakeDatabaseMetadata class to use new test function (to retrieve show command)